### ISSUE TITLE
Correct version in operator when OPERATOR_CONTAINER_VERSION is set

### ISSUE THREAD

**Issue by aljesusg** [kiali/kiali#3132]
When I upload a operator version from local I saw that in minikube show the version **v1.23.0-SNAPSHOT** when _OPERATOR_CONTAINER_VERSION=dev_ and the image loaded is localhost:5000/kiali/kiali-operator:dev .

I think that we should tag this with dev instead the latest one. What do you think ?

![Screenshot from 2020-08-18 10-28-23](https://user-images.githubusercontent.com/3019213/90489485-a4102e80-e13d-11ea-964f-6023c6150c67.png)

```yaml
# Source: kiali-operator/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: kiali-operator
  namespace: kiali-operator
  labels:
    helm.sh/chart: kiali-operator-1.23.0-SNAPSHOT
    app: kiali-operator
    app.kubernetes.io/name: kiali-operator
    app.kubernetes.io/instance: kiali-operator
    version: "v1.23.0-SNAPSHOT"
    app.kubernetes.io/version: "v1.23.0-SNAPSHOT"
    app.kubernetes.io/managed-by: Helm
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: kiali-operator
      app.kubernetes.io/instance: kiali-operator
  template:
    metadata:
      name: kiali-operator
      namespace: kiali-operator
      labels:
        # required for the operator SDK metric service selector
        name: kiali-operator
        helm.sh/chart: kiali-operator-1.23.0-SNAPSHOT
        app: kiali-operator
        app.kubernetes.io/name: kiali-operator
        app.kubernetes.io/instance: kiali-operator
        version: "v1.23.0-SNAPSHOT"
        app.kubernetes.io/version: "v1.23.0-SNAPSHOT"
        app.kubernetes.io/managed-by: Helm
      annotations:
        prometheus.io/scrape: "true"
    spec:
      serviceAccountName: kiali-operator
      containers:
      - name: operator
        image: "localhost:5000/kiali/kiali-operator:dev"
        imagePullPolicy: Always
        args:
        - "--zap-level=info"
        volumeMounts:
        - mountPath: /tmp/ansible-operator/runner
          name: runner
        env:
        - name: WATCH_NAMESPACE
          value: ""
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: OPERATOR_NAME
          value: kiali-operator
        - name: ANSIBLE_DEBUG_LOGS
          value: "true"
        - name: ANSIBLE_VERBOSITY_KIALI_KIALI_IO
          value: "1"
        ports:
        - name: http-metrics
          containerPort: 8383
        - name: cr-metrics
          containerPort: 8686
      volumes:
      - name: runner
        emptyDir: {}
      affinity:
        {}

```

**Comment by jmazzitelli** [comment]
First, some background. This has to do with the way the Helm chart creates the labels. That value you see is actually the helm chart's defined AppVersion. So these labels (specifically the version label) is the version of the operator resources that were installed by the helm chart - by that I mean the version is actually the version of all the YAML for the operator (the Deployment, the ServiceAccount, the Roles, etc.

What you are seeing is also a very specific edge case -- a developer used the Makefile target `operator-create` to deploy the operator container image that was pushed via `cluster-push` target.

Users won't see what you are seeing when installing using [the installation guide instructions for Helm](https://kiali.io/documentation/latest/installation-guide/#_helm_chart), and users using OLM to install the operator won't see this either.

In other words, this edge case is specific only to Kiali developers. I'd rather not code up special edge-case things in the helm chart to get this to show "dev" just for developers to see (developers should know what version they have pushed into their cluster via the makefile :)) Seeing `-SNAPSHOT` in their version should be enough for developers to see they have a locally built version on their machine.

**Comment by aljesusg** [comment]
ok in that case we can close this one :) thanks @jmazzitelli
